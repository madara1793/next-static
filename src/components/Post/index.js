// @flow
/* eslint-disable react/no-danger */

import React from 'react';
import Link from 'next/link';
import styled from 'styled-components';
import { distanceInWordsToNow } from 'date-fns';
import renderMarkup from '../../lib/renderMarkup';
import { type Post } from '../../../types/post';
import config from '../../../config';

const BlogPost = styled.article`
  background: ${config.theme.colors.main};
  pre[class*=language-] {
    ${'' /* padding: 0; */}
  }
`;

export default ({ title, author, date, tags, body, slug }: Post) =>
  <BlogPost itemScope itemType="http://schema.org/BlogPosting" className="post">
    <header>
      <Link href={`/post?post=${slug}`} as={`/post/${slug}`}>
        <a><h1 itemProp="headline" className="post--title">{title}</h1></a>
      </Link>
      <footer className="post--info">
        <span>
          <time itemProp="datePublished" dateTime={date}>
            {distanceInWordsToNow(date, { addSuffix: true })}
          </time>
        </span>
        <span itemProp="author">{author}</span>
      </footer>
    </header>
    <div
      className="post--body"
      dangerouslySetInnerHTML={{ __html: renderMarkup(body) }}
    />
    <footer>
      <small className="post--tags">
        <span>Filed under: </span>
        {tags.map((tag, index) =>
          <span key={tag} itemProp="keywords" className="post--tag">
            <Link
              href={`/tag?tag=${tag}`}
              as={`/tag/${tag.replace(/\s+/g, '-').toLowerCase()}`}
            >
              <a>{tag}</a>
            </Link>
            {index !== tags.length - 1 ? ', ' : ''}
          </span>
        )}
      </small>
    </footer>
  </BlogPost>;
