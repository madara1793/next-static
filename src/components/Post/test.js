import React from 'react';
import { renderToStaticMarkup as render } from 'react-dom/server';
import dom from 'cheerio';
import Post from './index';
import renderer from 'react-test-renderer';
// Global Test Setup
const Component = Post;

const defaultProps = {
  title: 'Hello',
  author: 'User',
  date: '7/22/2017',
  tags: ['javascript', 'angular'],
  body: 'lorem ipsum is the name making tests is this game',
  slug: 'test-post',
};

const makeProps = ({ title, author, date, tags, body, slug } = defaultProps) =>
  Object.assign(
    {},
    {
      title,
      author,
      date,
      tags,
      body,
      slug,
    }
  );

const loadComponent = el => dom.load(render(el));

describe('<Post /> with no args', () => {
  it('should render a post with default args', () => {
    const props = makeProps();
    const re = RegExp(props);

    const el = <Component {...props} />;

    const $ = loadComponent(el);

    const output = $('.post').html();
    const actual = re.test(output);
    const expected = true;

    expect(actual).toEqual(expected);
  });
});

describe('Snapshot::<Post />s', () => {
  it('should render the contents of the component.', () => {
    const props = makeProps();
    const el = <Component {...props} />;

    const tree = renderer.create(el).toJSON();

    expect(tree).toMatchSnapshot();
  });
});
