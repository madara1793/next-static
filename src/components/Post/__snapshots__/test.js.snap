// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Snapshot::<Post /> with default args should render our post 1`] = `
"<header><a href=\\"/post/test-post\\"><h1 itemprop=\\"headline\\" class=\\"post--title\\">Hello</h1></a><footer class=\\"post--info\\"><span><time itemprop=\\"datePublished\\" datetime=\\"7/22/2017\\">about 1 month ago</time></span><span itemprop=\\"author\\">User</span></footer></header><div class=\\"post--body\\"><p>lorem ipsum is the name making tests is this game</p>
</div><footer><small class=\\"post--tags\\"><span>Filed under: </span><span itemprop=\\"keywords\\" class=\\"post--tag\\"><a href=\\"/tag/javascript\\">javascript</a>, </span><span itemprop=\\"keywords\\" class=\\"post--tag\\"><a href=\\"/tag/angular\\">angular</a></span></small></footer>"
`;
